sentry_streams/sentry_streams/adapters/arroyo/consumer.py

```diff
@@ -59,9 +60,26 @@ def build_strategy(self, commit: Commit) -> ProcessingStrategy[Any]:
         follow.
         """
 
-        def add_route(message: Message[KafkaPayload]) -> RoutedValue:
-            value = message.payload.value
-            return RoutedValue(route=Route(source=self.source, waypoints=[]), payload=value)
+        def add_route(message: Message[KafkaPayload]) -> Union[FilteredPayload, RoutedValue]:
+            filtered = False
+            if self.header_filter:
+                headers: Headers = message.payload.headers
+                if self.header_filter not in headers:
+                    filtered = True
+
+            if filtered:
+                return FilteredPayload()
+            else:
+                value = message.payload.value
+                try:
+                    schema: Codec[Any] = get_codec(self.stream_name)
```

Comment:
I think you should not need to run the `get_codec` function for every message. I'd get the instance of the schema only once and append it as a flyweight object to all messages.

Reply:
should be fixed now